src/models/converters/safes.rs

```diff
@@ -29,6 +43,29 @@ impl SafeInfo {
                 .await,
             guard: info_provider.to_address_ex_optional(&self.guard).await,
             version: self.version.to_owned(),
+            implementation_version_state,
         }
     }
 }
+
+pub(super) fn calculate_version_state(
+    safe_version: &str,
+    min_chain_version: &Option<String>,
+) -> ImplementationVersionState {
+    if let Some(min_chain_version) = min_chain_version {
+        let sem_ver_safe = Version::parse(safe_version);
+        let sem_ver_min = Version::parse(min_chain_version);
+
+        if sem_ver_min.is_err() || sem_ver_safe.is_err() {
+            return ImplementationVersionState::Unknown;
+        }
+
+        return match sem_ver_safe.unwrap().cmp(&sem_ver_min.unwrap()) {
+            Ordering::Less => ImplementationVersionState::Outdated,
+            Ordering::Equal => ImplementationVersionState::UpToDate,
+            Ordering::Greater => ImplementationVersionState::UpToDate,
+        };
+    }
```

Comment:
Rust question: is there a way to have the `None` matching first and then dealing with the case of having a value (this would reduce branching in the function).

```rust
pub(super) fn calculate_version_state(
    safe_version: &str,
    min_chain_version: &Option<String>,
) -> ImplementationVersionState {
    if min_chain_version.is_none() {
        return ImplementationVersionState::Unknown
    }
    // from here on `min_chain_version` is `Some(T)`
}

Reply:
Yeah, you can absolutely do that. But in my implementation I am taking advantage of 2 features of rust, de-structuring and name shadowing (well the later, not sure if a "feature" is a bit of a strong word).

By doing 
```rust
let value = Some(1); // optional value

if let Some(value) = value {
   assert_eq!(value, 1);
}
```

I am reusing the variable name, while at the same time obtaining access to the value within the `Option<T>` if it is a some. the `if let` block is skipped if `value` turns out to be a `None`. You can also have `else` for `if let`s . If you want I can enclose the outer path in an `else` if it brings clarity to this.

Reply:
Ok what I was thinking can be accomplished with:

```rust
pub(super) fn calculate_version_state(
    safe_version: &str,
    min_chain_version: &Option<String>,
) -> ImplementationVersionState {
    // Deal with the exceptional case first
    let min_chain_version = match min_chain_version {
        Some(min_chain_version) => min_chain_version,
        None => return ImplementationVersionState::Unknown,
    };

    let sem_ver_safe = Version::parse(safe_version);
    let sem_ver_min = Version::parse(min_chain_version);

    if sem_ver_min.is_err() || sem_ver_safe.is_err() {
        return ImplementationVersionState::Unknown;
    }

    return match sem_ver_safe.unwrap().cmp(&sem_ver_min.unwrap()) {
        Ordering::Less => ImplementationVersionState::Outdated,
        Ordering::Equal => ImplementationVersionState::UpToDate,
        Ordering::Greater => ImplementationVersionState::UpToDate,
    };
}
```

Honestly in terms of readability I have no preference!